### DISCUSSION TITLE
Error when tokenizing Stripe element card

### DISCUSSION THREAD

**Discussion by y3owk1n** [swellstores/community#66] [Help] [answered]
While i am implementing my custom checkout flow, when I call the following code, I've got this error `"Missing value for stripe.confirmCardSetup intent secret: value should be a client_secret string."`

```
            await swell.payment.tokenize({
                card: {
                    onError: (err) => {
                        // inform the customer there was an error
                        console.log({ err });
                    },
                    onSuccess: async () => {
                        console.log("success");
                        //finally submit the form
                    },
                },
            });
```

I've done the integration from the dashboard as follows.

![image](https://user-images.githubusercontent.com/62775956/210229075-4088c48e-7ad2-4a3e-b218-c607db0c98a5.png)

Would like to check if I should include any client secret anywhere?

**Comment by y3owk1n** [comment]
Upon debuging this, I think it might be a bug from swell js on handling camelCase. As illustrated by the breakpoint image below, it seems like the response from authorization are camelCase, but the next function is using snakecase instead.

![image](https://user-images.githubusercontent.com/62775956/210254307-cbe879a3-daf4-438e-870c-9bfd188cb193.png)

**Comment by Buupu** [comment]
Experiencing the same issue. Very annoying as I've implemented my app with camelCase but looks like I'll have to go through and change everything to snakecase 😭

**Reply by y3owk1n** [reply]
@Buupu I think this might be a bug and had shared my debugging process [here](https://github.com/swellstores/swell-js/issues/79#issue-1517496033). I refuse to convert my codebase back to snake case and hope it could get a hotfix as soon as possible 😢

**Reply by Buupu** [reply]
Thanks for sharing! I'm tempted to patch it locally as well but don't really want to be forking an external lib. 

Will keep an eye on your post, hopefully they fix it soon 😭😭

**Reply by y3owk1n** [reply]
@Buupu yeah ~ finger cross 🤞🏻.

**Reply by y3owk1n** [reply]
@Buupu I've implemented some changes on my fork of swelljs [here](https://github.com/y3owk1n/swell-js/tree/fix-stripe-camelcase-issue). If you're interested to use it or check on it, feel free to use it temporarily. Don't think it well differ much when it comes to the official fix from swell team particularly on this issue.

**Reply by swellmike** [reply]
Thanks @y3owk1n !  I'll forward this patch to the Swell team for review and inclusion into the official SDK.

I think there may be further improvements, such as factoring in `options.useCamelCase` to transform the Stripe key.

**Comment by swellmike** [comment, marked as answer]
Try pulling down the latest swell-js (v3.18.3)
https://www.npmjs.com/package/swell-js

@y3owk1n The final patch looks slightly different than the original submission, so let us know if it works for you.

**Reply by y3owk1n** [reply]
@swellmike it works perfectly. I think the changes was pretty similar and did not affect anything at all. Thanks for the quick fix!
